extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -1089,7 +1089,7 @@ async function setupDeleteExcludeTest(proj: Project): Promise<[FileProjectEntry,
 
 	// confirm setup
 	should(proj.files.length).equal(3, 'number of file entries');
-	should(proj.folders.length).equal(2, 'number of folder entries');
+	should(proj.folders.length).equal(3, 'number of folder entries');
```

Comment:
what's the 3rd folder?

Reply:
It's for the external folder (`..`) that's technically included in the project (hence why it shows up here).  When constructing the project tree, external folders are excluded, and external files are automatically parented to the project root node.

Reply:
oh that's kinda weird. Was that a recent change that made the external folder now be counted in the project's folders? And the external folder got included when `../anotherScript.sql` was added to the project? I thought the folder only got added automatically when adding a file inside the project's folder for legacy style projects.

Reply:
> Was that a recent change?

So, because of how DacFx.Projects handles tracking folder contents and relative paths, it knows about external folders - so that's why it's showing up in the `SqlProject.Folders` collection over in DacFx (been like that since the beginning of that library).

Right now, we wait until generating the tree to filter out those folders and re-parent the external files to the root project node, since that's the first time that it's necessary to.  We could also shift that logic down to `project.ts:readFolders()` in ADS or even `HandleGetFoldersRequest()` in STS.

> the external folder got included when `../anotherScript.sql` was added to the project?

Yup

> I thought the folder only got added automatically when adding a file inside the project's folder for legacy-style projects

Yup, that's correct... for what gets written to the .sqlproj file.  DacFx.Projects's list of folders is a logical list, so it includes all the implied parents of all the files in addition to the folders that are explicitly listed in the .sqlproj.

Reply:
my main concern with the external folder getting automatically included is for SDK-style projects it could include any other .sql files that are also in that external folder, even if only one specific .sql file in that external folder was intended to be added to the project. Is that what would happen or am I overthinking it?

Reply:
Kim and I chatted offline about this.  Because DacFx processes the glob include/removes for files in SDK-style projects independently from folders then generates the list of folders from them, the correct files will and will not be included.  However, we decided that the logic to filter out external folders makes more sense in `project.readFolders()` to head that off as soon as possible, rather than when constructing the project tree from said list of folders.